src/templates/partials/source-file.html

```diff
@@ -0,0 +1,172 @@
+
+<!-- Render authors -->
+{% macro render_authors(authors) %}
+  {% set git_authors = config.plugins.get("git-authors") %}
+  <span class="md-source-file__fact">
+    <span class="md-icon" title="{{ lang.t('source.file.contributors') }}">
+      {% if authors | length == 1 %}
+        {% include ".icons/material/account.svg" %}
+      {% else %}
+        {% include ".icons/material/account-group.svg" %}
+      {% endif %}
+    </span>
+    <nav>
+      {% for author in authors %}
+        {%- if git_authors.config.show_email_address %}
+        <a href="mailto:{{ author.email }}">
+          {{- author.name -}}
+        </a>
+        {%- else -%}
+          {{- author.name -}}
+        {% endif -%}
+        {%- if loop.revindex > 1 %}, {% endif -%}
+      {% endfor %}
+    </nav>
+  </span>
+{% endmacro %}
+
+<!-- ---------------------------------------------------------------------- -->
+
+<!-- Render committers from GitHub -->
+{% macro render_committers_github(title) %}
+  <span class="md-icon" title="{{ lang.t('source.file.contributors') }}">
+    {% include ".icons/material/github.svg" %}
+  </span>
+  <span>{{ title }}</span>
+{% endmacro %}
+
+<!-- Render committers from GitLab -->
+{% macro render_committers_gitlab(title) %}
+  <span class="md-icon" title="{{ lang.t('source.file.contributors') }}">
+    {% include ".icons/material/gitlab.svg" %}
+  </span>
+  <span>{{ title }}</span>
+{% endmacro %}
+
+<!-- Render committers -->
+{% macro render_committers(authors) %}
+  <span class="md-source-file__fact">
+    {% if committers_source == "gitlab" %}
+      {{ render_committers_gitlab("GitLab") }}
+    {% else %}
+      {{ render_committers_github("GitHub") }}
+    {% endif %}
+    <nav>
+      {% for author in authors[:4] %}
+        <a
+          href="{{ author.url }}"
+          class="md-author"
+          title="@{{ author.login }}"
+        >
+          {% set separator = "&" if "?" in author.avatar else "?" %}
+          <img
+            src="{{ author.avatar }}{{ separator }}size=72"
+            alt="{{ author.name or 'GitHub user' }}"
+          />
+        </a>
+      {% endfor %}
+
+      <!-- More authors -->
+      {% set more = authors[4:] | length %}
+      {% if more > 0 %}
+        {% if page.edit_url %}
+          <a
+            href="{{ page.edit_url | replace('edit', 'blob') }}"
+            class="md-author md-author--more"
+          >
+            +{{ more }}
+          </a>
+        {% else %}
+          <span class="md-author md-author--more">
+            +{{ more }}
+          </span>
+        {% endif %}
+      {% endif %}
+    </nav>
+  </span>
+{% endmacro %}
+
+<!-- ---------------------------------------------------------------------- -->
+
+<!-- Determine date of last update -->
+{% if page.meta %}
+  {% if page.meta.git_revision_date_localized %}
+    {% set updated = page.meta.git_revision_date_localized %}
+  {% elif page.meta.revision_date %}
+    {% set updated = page.meta.revision_date %}
```

Comment:
I just noticed that we've been feeding Sphinx' `last_updated` context into this template.
https://github.com/jbms/sphinx-immaterial/blob/63a80bb381f1673e827939e8dc8f032b943bb80a/sphinx_immaterial/nav_adapt.py#L807
But this doesn't actually describe the last revision date. All pages have the last build date used even if the doc wasn't actually altered on the date it was built. This inaccuracy is not caused by the shallow checkout used in CI builds; I'm also seeing the same behavior locally.

From yesterday's RTD build for this PR (in [task_lists.html](https://sphinx-immaterial--338.org.readthedocs.build/en/338/task_lists.html)):
![image](https://github.com/jbms/sphinx-immaterial/assets/14963867/57198510-e1d4-4e16-a646-cb0dcd95fdd3)


This template feature would be better served by #341, but I guess this behavior would suffice since it has been this way. We could instead manually get the `os.stats(doc2path(pagename).st_birthdate`, but python docs say this info may not be available on all systems. And again, the checkout date is not the same as the birthdate since `git checkout` doesn't persist file attributes.

Reply:
Git doesn't store any times at the per-file level, and `git checkout` / `git clone` sets the modification time to the time of the checkout (for an existing work tree, only the modified files get updated time stamps).  Therefore we have to explicitly query git to find the most recent commit that modified the file as in #341, because the only timestamps stored by git are commit timestamps.  For that we will need a non-shallow checkout but that probably isn't too big of a deal in practice.

Reply:
I ran across a reference in upstream docs that recommends sparce checkouts. But I didn't see it get used in their CI (at least in community edition docs build).